design/cluster-api-provider-metal3/capm3-remediation-controller-improvement-proposal.md

```diff
@@ -0,0 +1,109 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# capm3-remediation-controller-improvement
+
+## Status
+
+implemented
+
+## Summary
+
+The original reboot remediation controller proposal [1] misses details on 
+on how remediation should actually be done. During implementation there was 
+some discussion if the Node should be deleted or not [2]. The decision was 
+to keep it simple and skip Node deletion.
+Skipping Node deletion has a big drawback though: workloads on unresponsive 
+Nodes won't be rescheduled quickly, because they are assumed to be still 
+running. Deleting the the Node signals that the workloads are not running 
+anymore, which results in quickly rescheduled workloads with less downtime.
+
+We would like to add Node deletion to the existing remediation strategy.
+
+## Motivation
+
+### Goals
+
+* Quick rescheduling of workloads on failed Nodes by signaling that they 
+  are not running anymore by deleting the Node.
+
+### Non-Goals
+
+* Change the remediation API 
+
+## Proposal
+
+The remediation controller's reconcile method will be modified to not only 
+reboot the Machine, but also delete the Node.
+
+### User Stories
+
+#### Story 1
+
+As a user, I expect minimal downtime of my workloads in case if Node issues.
+
+## Design Details
+
+Unfortunately adding Node deletion to the controller's reconcile method is a
+bigger change in the implementation than it sounds, because the old 
+one-step fencing process (trigger reboot by setting the appropriate annotation)
+becomes a multiple step process, and after each step we need to wait for
+success before executing the next one:
+
+* power Machine off
+* delete the Node
```

Comment:
Is label/annotation backup & restore omitted from this list on purpose or/and should we note those steps too here?

Reply:
good point, added